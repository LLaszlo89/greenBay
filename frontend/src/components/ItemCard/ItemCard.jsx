import React from "react";
import Card from "@material-ui/core/Card";
import CardHeader from "@material-ui/core/CardHeader";
import CardActions from "@material-ui/core/CardActions";
import CardContent from "@material-ui/core/CardContent";
import Typography from "@material-ui/core/Typography";
import Button from "@material-ui/core/Button";
import { CardMedia } from "@material-ui/core";
import { Grid } from "@material-ui/core";
import { makeStyles } from "@material-ui/core/styles";
import { itemSold } from "../../redux/actions/itemsActions";
import { connect } from "react-redux";

const useStyles = makeStyles(() => ({
  root: {
    height: "300px",
    width: "350px",
    margin: "15px",
    background: "linear-gradient(45deg,#D6FFD4  30%, #A8FF5F 70%)",
  },
  img: { height: "100px" },
  button: {
    fontFamily: "georgia",
    fontSize: "20px",
    margin: 0,
  },
}));

const ItemCard = (props) => {
  const classes = useStyles();
  const { id, title, price, description, URL } = props;

  const handelBuy = (id) => {
    props.buyId(id);
  };

  return (
    <Grid container justify="space-around" alignItems="center">
      <Card className={classes.root}>
        <CardHeader title={title} subheader={price + " green_$"} />
        <CardMedia className={classes.img} image={URL} />
        <CardActions>
          <Button
            className={classes.button}
            onClick={() => {
              handelBuy(id);
            }}
            size="small"
          >
            BUY NOW
          </Button>
        </CardActions>
        <CardContent>
          <Typography>{description}</Typography>
        </CardContent>
      </Card>
    </Grid>
  );
};

const mapDispatchToProps = (dispatch) => {
  return {
    buyId: (id) => dispatch(itemSold(id)),
  };
};

export default connect(null, mapDispatchToProps)(ItemCard);
