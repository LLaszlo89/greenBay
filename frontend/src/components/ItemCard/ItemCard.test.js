import React from "react";
import renderer from "react-test-renderer";
import ItemCard from "./ItemCard";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import configureStore from "redux-mock-store";

describe("Connect component to redux store ", () => {
  const mockStore = configureStore([]);


  const store = mockStore( [] );

  it("ItemCard Component should render correctly", () => {
    const tree = renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <ItemCard />
        </Provider>
      </BrowserRouter>
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
