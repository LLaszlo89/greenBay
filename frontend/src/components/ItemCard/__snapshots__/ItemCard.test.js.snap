// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Connect component to redux store  ItemCard Component should render correctly 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-around"
>
  <div
    className="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      className="MuiCardHeader-root"
    >
      <div
        className="MuiCardHeader-content"
      >
        <span
          className="MuiTypography-root MuiCardHeader-subheader MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
        >
          undefined green_$
        </span>
      </div>
    </div>
    <div
      className="MuiCardMedia-root makeStyles-img-2"
    />
    <div
      className="MuiCardActions-root MuiCardActions-spacing"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-3 MuiButton-textSizeSmall MuiButton-sizeSmall"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          BUY NOW
        </span>
      </button>
    </div>
    <div
      className="MuiCardContent-root"
    >
      <p
        className="MuiTypography-root MuiTypography-body1"
      />
    </div>
  </div>
</div>
`;
